### ISSUE TITLE
Git pull not working

### ISSUE THREAD

**Issue by sinrise** [bliker/cmder#419]
No idea what's going on here but 'git pull' returns:
    
    git: 'pull' is not a git command. See 'git --help'.

yet if you type:
    
    git --help

'pull' is right there in the list of commands... wtf?

**Comment by nihilon** [comment]
Did you compile git yourself? I suspect this isn't an issue with CMDer.
The help file will list the command even if it isn't accessible for some reason. The two are not related in that way.

What git is telling you is that the pull command is not in your path. This is for OS X, but the answer may help, especially if you compiled your copy of git: ["git pull" broken](http://stackoverflow.com/a/1466029).

Have you tried reinstalling git yet?

**Comment by nihilon** [comment]
So, the issue is probably the same issue in the answer I linked in my last comment. Git simply didn't install correctly, so this environment variable is missing:

```
GIT_EXEC_PATH=/opt/local/libexec/git-core
```

Both of the following cite that as being a possible issue, and so does the original answer I linked:

[Cygwin git pull not a command](http://superuser.com/q/198402)
[Cygwin git-pull not a command](http://stackoverflow.com/q/2399273)

That last one here, it's also stated that a clean reinstall solved the problem. If you want to avoid a reinstall you might also try locating or creating your `~/.bashrc` and entering the following lines:

```
export GIT_EXEC_PATH=/opt/local/libexec/git-core
export PATH=$GIT_EXEC_PATH:$PATH
```

**Comment by MartiUK** [comment]
@sinrise What version are you running? do you have git already installed and on your PATH already?

**Comment by nihilon** [comment]
@MartiUK They wouldn't have gotten the `git: 'pull' is not a git command...` error if git wasn't installed and on path. They would've gotten a `git: command not found` with mintty, bash, etc, or `'git is not recognized as an internal or external command, operable program or batch file` under the Windows Command Processor (cmd) if it wasn't installed and on path.

**Comment by sinrise** [comment]
I was able to fix this issue but editing init.bat in the [Cmder install folder]/vendor folder

Change:

    @set git_install_root=%CMDER_ROOT%\vendor\msysgit

To:

    @set git_install_root=%SYSTEMROOT%\Program Files (x86)\Git\bin

**Comment by MartiUK** [comment]
@nihilon That's what I meant by having git already installed.

**Comment by nihilon** [comment]
@MartiUK Got ya. Good to know for future reference.

**Comment by glucas** [comment]
I'm seeing this same problem with a fresh install of Cmder v1.2, I'm using the version of msysgit packaged with Cmder:

λ set git
GIT_INSTALL_ROOT=C:\opt\cmder\vendor\msysgit

λ where git
C:\opt\cmder\vendor\msysgit\bin\git.exe

**Comment by glucas** [comment]
Well -- answering my own question, I compared that libexec\git-core with my old cmder 1.1.4 installation, and basically all the commands are missing.

**Comment by UncleGus** [comment]
I had this issue and a clean install did not fix it, nor did setting environment variables. My laptop was not affected, so I compared files in the libexec folders. The laptop had 163 while the pc only had 131. I copied all the files over from the laptop to the pc and hey presto, fixed.

The pc was able to clone my repo and retrieve it the first time, and push at least once and then mysteriously became broken after that, and when I re-installed git, the repo was already registered, so perhaps it wasn't a completely clean install.

The pc is running windows 10 while the laptop is on windows 7.
